Create pickup mechanic

Isa1997 opened the pull request:
Create pickup mechanic
Add inventory render

Isa1997 committed:
Create pickup mechanich
Add inventory render

gavrilovmiroslav commented on scripts/gameplay/Library.lua:
```diff
@@ -35,4 +35,12 @@ UpdateDijkstra = ECS.Component()
+FollowBehaviour = ECS.Component{ distance = 5 }
+
+Inventory = ECS.Component{items = {}}
+InventoryEntity = World:Entity(Inventory{items = {}})
```
Instead of this, add Inventory to PlayerEntity and any other thing you want to carry items

gavrilovmiroslav replied:
When you want to open the player's inventory, query it directly via `PlayerEntity[Inventory]` and if you want some other inventory, get their entities and do the same :)

gavrilovmiroslav commented on scripts/gameplay/render/InventoryRender.lua:
```diff
@@ -0,0 +1,40 @@
+end
+
+function InventoryRender:Restore()
+	local widget = Widgets.Inventory
```
Maybe we should discuss rendering in layers?

gavrilovmiroslav commented on scripts/gameplay/render/InventoryRender.lua:
```diff
@@ -0,0 +1,40 @@
+	end
+end
+
+Widgets = {
```
Why not just make components and render entities with Widget?

Isa1997 committed:
Addressed comments from review
- Move widgets from inventory
- Remove inventory entity and add inv component to player
- Make pickup mechanic more generic

Isa1997 committed:
Merge with main

Isa1997 committed:
Merge with main

Isa1997 committed:
mege with main
